test/k8sT/Services.go

```diff
@@ -412,6 +412,93 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	Context("Checks local redirect policy", func() {
+		var (
+			deploymentYAML string
+			lrpServiceName string
+			app1IP         string
+			app2IP         string
+			be1IP          string
+			be2IP          string
+		)
+
+		BeforeAll(func() {
+			deploymentYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-test.yaml")
+			lrpServiceName = "lrp-demo-service"
+			res := kubectl.ApplyDefault(deploymentYAML)
+			res.ExpectSuccess("Unable to apply %s", deploymentYAML)
+			waitPodsDs()
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(deploymentYAML)
+		})
+
+		getSingletonPodIP := func(label string) string {
+			podsIPs, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, label)
+			Expect(err).To(BeNil(), "Cannot get pod IP for %s pod", label)
+			Expect(len(podsIPs)).To(Equal(1), "Got more than 1 ip for %s pod", label)
+			for _, ip := range podsIPs {
+				return ip
+			}
+			// Should never reach here
+			return ""
+		}
+
+		It("LRP connectivity", func() {
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, "cilium service list", "Cannot retrieve services on cilium pod")
+				service.ExpectContains("LocalRedirect", "LocalRedirect is not present in the cilium service list")
+			}
+
+			ciliumPodK8s1, err := kubectl.GetCiliumPodOnNode(helpers.K8s1)
+			ExpectWithOffset(2, err).Should(BeNil(), "Cannot get cilium pod on k8s1")
+			ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.K8s2)
+			ExpectWithOffset(2, err).Should(BeNil(), "Cannot get cilium pod on k8s2")
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, lrpServiceName)
+			Expect(err).To(BeNil(), "Cannot get svc IP")
+			httpSVCURL := fmt.Sprintf("http://%s/", clusterIP)
+			tftpSVCURL := fmt.Sprintf("tfpt://%s/hello", clusterIP)
+			app1IP = getSingletonPodIP("id=app1")
+			app2IP = getSingletonPodIP("id=app2")
+			be1IP = getSingletonPodIP("id=be1")
+			be2IP = getSingletonPodIP("id=be2")
+
+			By("Checking traffic goes to local backend")
+			monitorRes1, monitorCancel1 := kubectl.MonitorStart(ciliumPodK8s1)
+			testCurlFromPods("id=app1", httpSVCURL, 10, 0)
+			testCurlFromPods("id=app1", tftpSVCURL, 10, 0)
+			monitorCancel1()
+			monitorRes1.ExpectSuccess()
+			monitorRes1.ExpectContains(fmt.Sprintf("SrcIP=%s DstIP=%s", app1IP, be1IP))
```

Comment:
Alternatively, you could check against pod name. Echoserver (both - tcp and udp) should return `Hostname: ...` which contains a pod name.

Reply:
Thanks! Changed to check the returned pod name, I think that's better :)